metagraph/src/cli/transform_annotation.cpp

```diff
@@ -474,27 +483,70 @@ int transform_annotation(Config *config) {
         thread_pool.join();
         std::atomic_thread_fence(std::memory_order_acquire);
 
-        uint64_t min_cols = num_columns * config->intersect_ratio;
-        logger->trace("Selecting k-mers annotated in >= {} / {} columns",
-                      min_cols, num_columns);
+        logger->trace("Selecting k-mers annotated in {} <= * <= {} (out of {}) columns",
+                      min_cols, max_cols, num_columns);
 
-        sdsl::bit_vector intersection(sum.size(), false);
+        sdsl::bit_vector mask(sum.size(), false);
         for (uint64_t i = 0; i < sum.size(); ++i) {
-            if (sum[i] >= min_cols) {
-                intersection[i] = true;
+            if (sum[i] >= min_cols && sum[i] <= max_cols) {
+                mask[i] = true;
             }
         }
```

Comment:
you can use `call_nonzeros` from `vector_algorithm.hpp` to speed this up.

Reply:
hm, not sure that makes sense. It's just a single pass, relative to summing up all columns, this should be cheap.
Also, this won't work for zeros, so I'll have to write a special case for `min_cols = 0`

Reply:
In this case, if the user sets `min_cols = 0` and `max_cols = inf`, then there's no need to even read the annotators and compute `sum`, we can just initialize `mask` to be all ones